plugins/filter_log_to_metrics/log_to_metrics.c

```diff
@@ -913,18 +965,14 @@ static int cb_log_to_metrics_filter(const void *data, size_t bytes,
                         flb_ra_destroy(ra);
                         ra = NULL;
                     }
+                    ctx->new_data = FLB_TRUE;
```

Comment:
Instead of setting `ctx->new_data` on each "good" switch branch just set it after the switch block, the `default` branch exits immediately so there's no risk of false positives.